examples/add_new_model_to_data_labeler.ipynb

```diff
@@ -0,0 +1,419 @@
+    "            tf.keras.layers.Dense(num_labels, activation='softmax'))\n",
+    "\n",
+    "        # Output the model into a .pb file for TensorFlow\n",
+    "        argmax_layer = tf.keras.backend.argmax(self._model.output)\n",
+    "\n",
+    "        # Create confidence layers\n",
+    "        final_predicted_layer = CharacterLevelLstmModel._argmax_threshold_layer(\n",
+    "            num_labels, threshold=0.0, default_ind=default_ind)\n",
+    "\n",
+    "        argmax_outputs = self._model.outputs + \\\n",
+    "                         [argmax_layer,\n",
+    "                          final_predicted_layer(argmax_layer, self._model.output)]\n",
+    "        self._model = tf.keras.Model(self._model.inputs, argmax_outputs)\n",
+    "\n",
+    "        # Compile the model\n",
+    "        softmax_output_layer_name = self._model.outputs[0].name.split('/')[0]\n",
+    "        losses = {softmax_output_layer_name: \"categorical_crossentropy\"}\n",
+    "\n",
+    "        # use f1 score metric\n",
+    "        f1_score_training = F1Score(num_classes=num_labels, average='micro')\n",
+    "        metrics = {softmax_output_layer_name: ['acc', f1_score_training]}\n",
+    "\n",
+    "        self._model.compile(loss=losses,\n",
+    "                            optimizer=\"adam\",\n",
+    "                            metrics=metrics)\n",
+    "\n",
+    "        self._epoch_id = 0\n",
+    "        self._model_num_labels = num_labels\n",
+    "        self._model_default_ind = default_ind\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00e5b5c4",
+   "metadata": {},
+   "source": [
+    "## Integrate the LSTM model to the DataLabeler"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cf6ec4a5",
+   "metadata": {},
+   "source": [
+    "Once the LSTM model is built, it replaces the existing model in the DataLabeler pipeline, which is then trained on the given dataset. Note that, as the DataLabeler is trained on the above tabular dataset, its label mapping is updated by the list of column names in that dataset while training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "02e24493",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get labels from the given dataset\n",
+    "value_label_df = data_train.reset_index(drop=True).melt()\n",
+    "value_label_df.columns = [1, 0]  # labels=1, values=0 in that order\n",
+    "value_label_df = value_label_df.astype(str)\n",
+    "labels = value_label_df[1].unique().tolist()\n",
+    "\n",
+    "# build new data labeler with the new lstm model\n",
+    "data_labeler = dp.DataLabeler(labeler_type='structured', trainable=True)\n",
+    "model = CharacterLevelLstmModel(labels + ['UNKNOWN']) # extend the labels with the default label 'UNKNOW'\n",
+    "data_labeler.set_model(model)\n",
+    "\n",
+    "# set default label (one of the column names) to the data labeler pipeline\n",
+    "params = {'default_label': 'comment'}\n",
```

Comment:
Why do we have this and the above part with adding the UNKNOWN? Only one of these should be needed right?

Reply:
And can you please add a conclusion at the end?